components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/authz/OAuth2AuthzEndpoint.java

```diff
@@ -4141,7 +4141,13 @@ private Response forwardToOauthResponseJSP(OAuthMessage oAuthMessage, String par
             HttpServletRequest request = oAuthMessage.getRequest();
             request.setAttribute(USER_TENANT_DOMAIN, authenticatedUser.getTenantDomain());
             request.setAttribute(TENANT_DOMAIN, authorizationResponseDTO.getSigningTenantDomain());
-            request.setAttribute(SERVICE_PROVIDER, getServiceProvider(authorizationResponseDTO.getClientId()));
+            ServiceProvider serviceProvider = getServiceProvider(authorizationResponseDTO.getClientId());
+            if (serviceProvider != null) {
```

Comment:
Only a suggession. Fine with the current implementation. We can avoid code shifting by having like below.
if (serviceProvider != null && serviceProvider.getApplicationName() != null) {
    request.setAttribute(SERVICE_PROVIDER, serviceProvider.getApplicationName());
}

Reply:
Addressed this with the commit 3e83c5c5d95ec6442b6aaa3d13f69d9f9a92984f